client/src/modules/break_even_reference/modals/break_even_reference.modal.js

```diff
@@ -0,0 +1,80 @@
+angular.module('bhima.controllers')
+  .controller('BreakEvenReferenceModalController', BreakEvenReferenceModalController);
+
+BreakEvenReferenceModalController.$inject = [
+  '$state', 'BreakEvenReferenceService', 'NotifyService', 'appcache',
+];
+
+/**
+ * Break Even Reference Modal Controller
+ */
+
+function BreakEvenReferenceModalController($state, BreakEvenReference, Notify, AppCache) {
+  const vm = this;
+  const cache = AppCache('AccountReferenceModal');
+
+  vm.breakEvenReferences = {};
+  vm.stateParams = {};
+
+  // exposed methods
+  vm.submit = submit;
+  vm.closeModal = closeModal;
+  vm.onSelectAccountReference = onSelectAccountReference;
+
+  if ($state.params.creating || $state.params.id) {
+    cache.stateParams = $state.params;
+    vm.stateParams = cache.stateParams;
+  } else {
+    vm.stateParams = cache.stateParams;
+  }
+  vm.isCreating = vm.stateParams.creating;
+
+  if (!vm.isCreating) {
+    BreakEvenReference.read(vm.stateParams.id)
+      .then(data => {
+        vm.breakEvenReferences = data;
```

Comment:
Can we call this "breakEvenReference" (singular) or just "reference"?  It is confusing to make it plural since it is a single object.